pkg/ambassador/tap/stream/stream.go

```diff
@@ -224,3 +231,56 @@ func (s *Stream) open(ctx context.Context) error {
 	}
 	return nil
 }
+
+func (s *Stream) refresh(ctx context.Context) error {
+	err := s.TargetRegistry.Register(ctx, s.getTarget())
```

Comment:
I don't get this function, some description would be appreciated about the motives here.
For me it seems the Target gets saved into the registry (overriding any existing entry), then the list of Targets are fetched from the registry to compare the IP list. Why would they differ if the target got just saved into the registry based on the stream?

Reply:
Yes, the NSP overwrite the status to `DISABLED` for newly registered targets, so the tapa has to verify if it is still in the `ENABLED` status, if it is, then it is fine and the refresh worked on time, but if it is not, it would mean the target has to register a new time (so it call the open function).
The Register function is a function that takes registration and updates, so NSP would consider it as a new registration (just it will replace the status since `ENABLED` is not allowed for new registration).